src/astbuilder.nit

```diff
@@ -934,3 +934,205 @@ redef class AAnnotation
 		_n_args = n_args
 	end
 end
+
+redef class MEntity
+	# Return a ast representation of self model entity.
+	# This method simplifies the creation of the ast based on the information available in the model.
+	fun create_ast_representation(astbuilder: nullable ASTBuilder): ANode is abstract
```

Comment:
I feel like this method should actually be called `to_anode` or `to_ast` or something.

Reply:
Also, I still have no clue if I should bother building my own ASTBuilder or just pass `null`. Maybe the documentation should explain that?

Reply:
The original objective of ASTBuilder is to provide a context for typing, the problem is that in most cases this context is not used ... The idea of defining it as nullable is that if the user has this context, he provides it. But it's not really essential since in the majority of cases the astbuilder can be built with the information available in the model.